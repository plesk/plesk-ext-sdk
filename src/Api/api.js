// Copyright 1999-2025. WebPros International GmbH. All rights reserved.

import { stringify } from 'qs';

const request = (url, config = {}) => {
    if (window.Ajax) {
        window.Ajax.activeRequestCount++;
    }

    const tokenEl = document.getElementById('forgery_protection_token');
    config = {
        credentials: 'same-origin',
        ...config,
        headers: {
            'X-Forgery-Protection-Token': tokenEl ? tokenEl.content : null,
            'X-Requested-With': 'XMLHttpRequest',
            ...config.headers,
        },
    };

    return fetch(url, config)
        .then(response => response.json().then(data => {
            data.response = response;
            return data;
        }))
        .finally(() => {
            if (window.Ajax) {
                window.Ajax.activeRequestCount--;
            }
        });
};

const addParams = (url, params) => {
    params = stringify(params);

    if (params) {
        url += (url.indexOf('?') === -1 ? '?' : '&') + params;
    }

    return url;
};

export default {
    get: (url, params = {}) => request(addParams(url, params), { method: 'GET' }),
    post: (url, params = {}) => request(url, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json; charset=utf-8',
        },
        body: JSON.stringify(params),
    }),
};
