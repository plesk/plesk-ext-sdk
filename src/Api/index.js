// Copyright 1999-2025. WebPros International GmbH. All rights reserved.

export { default as api } from './api';
export { default as ApiProvider } from './ApiProvider';
export { default as withApi } from './withApi';
