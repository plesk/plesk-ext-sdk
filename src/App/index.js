// Copyright 1999-2025. WebPros International GmbH. All rights reserved.

export { default } from './App';
