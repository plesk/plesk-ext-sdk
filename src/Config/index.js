// Copyright 1999-2025. WebPros International GmbH. All rights reserved.

export { default as ConfigProvider } from './ConfigProvider';
export { default as withConfig } from './withConfig';
