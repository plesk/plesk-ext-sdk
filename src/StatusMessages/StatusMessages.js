// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { PureComponent, Toaster, PropTypes } from '@plesk/ui-library';
import StatusMessagesProxy from './StatusMessagesProxy';
import StatusMessagesProvider from './StatusMessagesProvider';
import { withRouter } from 'react-router-dom';

export class StatusMessages extends PureComponent {
    static propTypes = {
        children: PropTypes.node.isRequired,
        location: PropTypes.shape({
            pathname: PropTypes.string.isRequired,
        }).isRequired,
        statusMessages: PropTypes.shape({
            clear: PropTypes.func.isRequired,
            setToaster: PropTypes.func.isRequired,
        }),
    };

    static defaultProps = {
        statusMessages: new StatusMessagesProxy(),
    };

    componentDidUpdate(prevProps) {
        if (prevProps.location.pathname !== this.props.location.pathname) {
            this.props.statusMessages.clear();
        }
    }

    render() {
        return (
            <StatusMessagesProvider value={this.props.statusMessages}>
                {this.props.children}
                <Toaster ref={this.props.statusMessages.setToaster} />
            </StatusMessagesProvider>
        );
    }
}

export default withRouter(StatusMessages);
