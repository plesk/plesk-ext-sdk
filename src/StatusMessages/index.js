// Copyright 1999-2025. WebPros International GmbH. All rights reserved.

export { default } from './StatusMessages';
export { default as StatusMessagesProvider } from './StatusMessagesProvider';
export { default as StatusMessagesProxy } from './StatusMessagesProxy';
export { default as withStatusMessages } from './withStatusMessages';
