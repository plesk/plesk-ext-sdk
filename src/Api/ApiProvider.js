// Copyright 1999-2025. WebPros International GmbH. All rights reserved.

import { createElement, createContext } from 'react';
import PropTypes from 'prop-types';
import api from './api';

export const ApiProviderContext = createContext();

const ApiProvider = ({ children, value = api }) => (
    <ApiProviderContext.Provider value={value}>
        {children}
    </ApiProviderContext.Provider>
);

ApiProvider.propTypes = {
    children: PropTypes.node.isRequired,
    value: PropTypes.object,
};

export default ApiProvider;
