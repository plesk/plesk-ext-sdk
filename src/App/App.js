// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { LocaleProvider } from '@plesk/ui-library';
import PropTypes from 'prop-types';
import { createElement } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { ConfigProvider } from '../Config';
import { ApiProvider } from '../Api';
import StatusMessages from '../StatusMessages';
import Page from '../Page';

const App = ({ locale, routes, ...props }) => (
    <ConfigProvider
        value={{
            baseUrl: props.baseUrl,
        }}
    >
        <LocaleProvider messages={locale}>
            <ApiProvider>
                <StatusMessages>
                    <BrowserRouter basename={props.baseUrl}>
                        <Routes>
                            {routes.map(({ path, component: Tag, ...routeProps }) => (
                                <Route
                                    key={path}
                                    path={path}
                                    render={() => <Page path={path} routes={routes}><Tag {...props} /></Page>}
                                    {...routeProps}
                                />
                            ))}
                        </Routes>
                    </BrowserRouter>
                </StatusMessages>
            </ApiProvider>
        </LocaleProvider>
    </ConfigProvider>
);

App.propTypes = {
    routes: PropTypes.array.isRequired,
    locale: PropTypes.object.isRequired,
    baseUrl: PropTypes.string.isRequired,
};

export default App;
