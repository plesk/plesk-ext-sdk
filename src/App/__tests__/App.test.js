// Copyright 1999-2018. Plesk International GmbH. All rights reserved.
import { createElement } from 'react';
import App from '../App';
import { ConfigProvider } from '../../Config';
import { ApiProvider } from '../../Api';
import StatusMessages from '../../StatusMessages';
import { shallow } from 'enzyme';

describe('App', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <App
                baseUrl=""
                locale={{}}
                routes={[{
                    path: '/',
                    componentPath: 'Index',
                    component: () => null,
                }]}
            />,
        );

        expect(wrapper.find(ConfigProvider)).toHaveLength(1);
        expect(wrapper.find(ApiProvider)).toHaveLength(1);
        expect(wrapper.find(StatusMessages)).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });
});
