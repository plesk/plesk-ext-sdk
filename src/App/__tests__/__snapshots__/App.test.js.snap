// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`App renders correctly 1`] = `
<ConfigProvider
  value={
    {
      "baseUrl": "",
    }
  }
>
  <LocaleProvider
    messages={{}}
  >
    <ConfigProvider
      value={
        {
          "get": [Function],
          "post": [Function],
        }
      }
    >
      <withRouter(StatusMessages)>
        <BrowserRouter
          basename=""
        >
          <Routes>
            <Route
              componentPath="Index"
              key="/"
              path="/"
              render={[Function]}
            />
          </Routes>
        </BrowserRouter>
      </withRouter(StatusMessages)>
    </ConfigProvider>
  </LocaleProvider>
</ConfigProvider>
`;
